compiler/parser/chapel.lex

```diff
@@ -468,6 +487,63 @@ static const char* eatStringLiteral(yyscan_t scanner, const char* startChar) {
   return astr(stringBuffer);
 }
 
+static const char* eatMultilineStringLiteral(yyscan_t scanner,
+                                             const char* startChar) {
+  YYLTYPE*   yyLloc  = yyget_lloc(scanner);
+  const char startCh = *startChar;
+  int startChCount = 0;
+  int        c       = 0;
+
+  newString();
+
+  while (((c = getNextYYChar(scanner)) != startCh || startChCount < 2) && c != 0) {
+    if (c == startCh) {
+      startChCount++;
+    } else {
+      startChCount = 0;
+    }
+
+    if (c == '\"') {
+      // escape double quotes
+      addCharEscape('\\');
+      addCharEscape('"');
+    } else if (c == '?') {
+      // backslash escape ? to avoid C trigraphs
+      addCharEscape('\\');
+      addCharEscape('?');
+    } else if (c == '\n') {
+      // translate newline into two characters "\n"
+      addCharEscape('\\');
+      addCharEscape('n');
+    } else if (c == '\t') {
+      // translate tab into two characters "\t"
+      addCharEscape('\\');
+      addCharEscape('t');
+    } else if (c == '\\') {
+      // translate backslash into an escaped double-backslash
+      addCharEscape('\\');
+      addCharEscape('\\');
```

Comment:
If this is a list of escape sequences in C, it's incomplete. See https://en.wikipedia.org/wiki/Escape_sequences_in_C

it should also handle `\a` `\b` `\f` `\r` `\v`, shouldn't it?

Alternatively, maybe you can explain why we don't need to handle those? I could believe an argument that only `\` `"` and `\n` need to be handled - but then I don't see why you are handling tab.

Reply:
I'd prefer whatever logic is necessary here be moved to a new function named something like addCharEscape.